### PR TITLE
Add 'Flight Cursor' flying style option

### PR DESCRIPTION
**Pull request by Baezon**
Adds an in game option to switch to a different kind of flying, where the primary control input moves a sort of cursor which determines rotational speed in that direction. Additionally, mods can specify that ships can actually fire in the direction of the flight cursor.

https://github.com/scp-fs2open/fs2open.github.com/assets/17305613/466a43fc-0bd9-4101-a8df-db07ff3e2298

### CODE REVIEW

**Review by Goober5000 — APPROVED** [review]
Looks reasonable.  I'm not super familiar with the hud code though, so I wouldn't mind a second pair of eyes on this.

**Comment by Baezon** [comment]
External or free look now resets the flight cursor, and now a ship with '$Aims at flight cursor' will switch to that style for that ship, even if the user has 'ship locked' selected, otherwise a player might not even know the ship has the capability.

**Review by JohnAFernandez — APPROVED** [review]
Otherwise it looks good afaict!

File: code/playerman/playercontrol.cpp [hunk 1]
```diff
@@ -60,6 +61,59 @@ physics_info Descent_physics;			// used when we want to control the player like
...
 
+angles Player_flight_cursor;
+
+FlightMode Player_flight_mode = FlightMode::ShipLocked;
+
+auto FlightModeOption = options::OptionBuilder<FlightMode>("Game.FlightMode",
+	std::pair<const char*, int>{"Flight Mode", 1842},
+	std::pair<const char*, int>{"Choose the flying style to use during gameplay.", 1843})
+	.category(std::make_pair("Game", 1824))
+	.level(options::ExpertLevel::Advanced)
```
[the remark this answers is not in the document]
Comment (Baezon) [reply code/playerman/playercontrol.cpp:72]: What about its options? Extent/deadzone

[the remark this answers is not in the document]
Comment (JohnAFernandez) [reply code/playerman/playercontrol.cpp:72]: It's hard to say.  There's value in putting them right next to the toggle, but they are definitely not going to be used that often.  I would err towards keeping them together, but I'm open to both.

**Pull request merged**